pilot/pkg/serviceregistry/kube/controller/ambient/workloads_test.go

```diff
@@ -867,34 +1053,97 @@ func TestEndpointSliceWorkloads(t *testing.T) {
 		result []*workloadapi.Workload
 	}{
 		{
-			name:   "api server",
-			inputs: []any{},
+			name: "api server",
+			inputs: []any{
+				kubernetesApiServerService("1.2.3.4"),
+			},
+			slice: kubernetesApiServerEndpoint("172.18.0.5"),
+			result: []*workloadapi.Workload{{
+				Uid:         "cluster0/discovery.k8s.io/EndpointSlice/default/kubernetes/172.18.0.5",
+				Name:        "kubernetes",
+				Namespace:   "default",
+				Addresses:   [][]byte{netip.MustParseAddr("172.18.0.5").AsSlice()},
+				Network:     testNW,
+				Status:      workloadapi.WorkloadStatus_HEALTHY,
+				NetworkMode: workloadapi.NetworkMode_HOST_NETWORK,
+				ClusterId:   testC,
+				Services: map[string]*workloadapi.PortList{
+					"default/kubernetes.default.svc.domain.suffix": {
+						Ports: []*workloadapi.Port{{
+							ServicePort: 443,
+							TargetPort:  6443,
+						}},
+					},
+				},
+			}},
+		},
+		{
+			name: "pod as part of selectorless service without targetRef",
+			inputs: []any{
+				model.ServiceInfo{
+					Service: &workloadapi.Service{
+						Name:      "svc",
+						Namespace: "default",
+						Hostname:  "svc.default.svc.domain.suffix",
+						Ports: []*workloadapi.Port{
+							{
+								ServicePort: 80,
+								TargetPort:  80,
+							},
+						},
+					},
+					PortNames: map[int32]model.ServicePortName{
+						80: {PortName: "http"},
+					},
+					// no selector!
+					LabelSelector: model.LabelSelector{},
+					Source:        kind.Service,
+				},
+			},
+			// EndpointSlice manually created with the IP of the pod, but does NOT have a targetRef
 			slice: &discovery.EndpointSlice{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "kubernetes",
+					Name:      "svc-123",
 					Namespace: "default",
 					Labels: map[string]string{
-						discovery.LabelServiceName: "kubernetes",
+						discovery.LabelServiceName: "svc",
 					},
 				},
 				AddressType: discovery.AddressTypeIPv4,
 				Endpoints: []discovery.Endpoint{
 					{
-						Addresses: []string{"172.18.0.5"},
+						Addresses: []string{"1.2.3.4"},
 						Conditions: discovery.EndpointConditions{
 							Ready: ptr.Of(true),
 						},
 					},
 				},
 				Ports: []discovery.EndpointPort{
 					{
-						Name:     ptr.Of("https"),
+						Name:     ptr.Of("http"),
 						Protocol: ptr.Of(v1.ProtocolTCP),
-						Port:     ptr.Of(int32(6443)),
+						Port:     ptr.Of(int32(80)),
 					},
 				},
 			},
-			result: nil,
```

Comment:
IDK what this test was before... it did not have `Source: Kubernetes` so it was mistakenly dropping the result and not testing anything. Bad test; the fix is not related to this PR